component/src/shielded_pool/component.rs

```diff
@@ -168,31 +181,71 @@ impl Component for ShieldedPool {
 
     #[instrument(name = "shielded_pool", skip(self, ctx, tx))]
     async fn execute_tx(&mut self, ctx: Context, tx: &Transaction) {
-        let _should_quarantine = tx
-            .transaction_body
-            .actions
-            .iter()
-            .any(|action| matches!(action, Action::Undelegate { .. }));
+        let mut should_quarantine = tx.transaction_body.actions.iter().find_map(|action| {
+            if let Action::Undelegate(Undelegate {
+                validator_identity, ..
+            }) = action
+            {
+                Some(validator_identity)
```

Comment:
How do we know this `validator_identity` is unique?

Reply:
Cool, I remembered we'd talked about that, but I wasn't sure where the check is.

Not blocking on this PR, but I'm a little worried about how to manage the complexity of having different checks in different components, since it's hard to refer to "lines 846 to 848 in blahblah.rs".

In the IBC component, we did a refactor pass where we put each logical check into a separate function, so that the verification function reads as a list of explicit, named checks.

We don't do this in the SP or staknig component, for historical reasons. But if we refactored the code that way, we'd be able to put a comment in referring to a specific check by name, like
```rust
// We know there can only be one validator identity, because of
// stake::checks::undelegation_only_from_one_validator()
```

However, that's a bunch of refactoring (albeit mechanical and easy), so we should do it later.